test/distributed/_tensor/test_dtensor.py

```diff
@@ -763,6 +763,25 @@ def test_implicit_replication(self):
             self.assertEqual(local_shard.shape, (4, 3))
             self.assertEqual(local_shard, torch.ones(4, 3) + torch.ones(3))
 
+    @with_comms
+    def test_auto_implicit_replication(self):
+        mesh = init_device_mesh(self.device_type, (self.world_size,))
+
+        local_tensor = torch.ones(self.world_size, 3, device=self.device_type)
+        sharded_dtensor = DTensor.from_local(local_tensor, mesh, [Shard(0)])
+
+        # automatically turn tensor to DTensor replicate when ndim = 0 and numel = 1
+        ndim_0_tensor = torch.tensor(1, device=self.device_type)
+        self.assertEqual(
+            (sharded_dtensor + ndim_0_tensor).to_local(), local_tensor + ndim_0_tensor
+        )
+
+        # automatically turn tensor to DTensor replicate when ndim = 1 and numel = 1
+        numel_1_tensor = torch.tensor([1], device=self.device_type)
+        self.assertEqual(
```

Comment:
can you add tests where if it's a "Scalar" Tensor, we don't throw warning? i.e. self.assertNoWarn